### PR TITLE
student view + cleanup

### PR DESCRIPTION
**Pull request by Li-Evelyn**
* render student assignment list view
* render student assignment questions view
* bug fixes here and there
* warning cleanup

### CODE REVIEW

**Comment by majesticwhales** [comment]
ui test is looking good

**Review by majesticwhales — APPROVED** [review]
looks good to me  but lets get a second pair of eyes on it?

**Review by pwang14 — APPROVED** [review]
Not too familiar w react, but lgtm.

File: client/src/components/dashboard.js [hunk 1]
```diff
@@ -75,8 +72,13 @@ function Dashboard(props) {
         setActiveAssignment(a);
         localStorage.setItem("assignment_name", a.name)
         localStorage.setItem("assignment_id", a.id)
-        console.log(a.name)
-        navigate(`${getAssignmentLink(userType, activeCourse.name, a.name)}`)
+        navigate(`${getAssignmentLink(userType, a.name, a.name)}`)
```
Comment (TaliaWang) [inline client/src/components/dashboard.js:75]: should the first one be a course name?

**Review by TaliaWang — APPROVED** [review]

File: client/src/components/dashboard.js (hunk 1 above)
Comment (Li-Evelyn) [reply client/src/components/dashboard.js:75]: ayk yes you are correct

**Pull request merged**